source/lac/scalapack.cc

```diff
@@ -207,6 +211,112 @@ ScaLAPACKMatrix<NumberType>::copy_to (FullMatrix<NumberType> &matrix) const
 
 
 
+template <typename NumberType>
+void
+ScaLAPACKMatrix<NumberType>::copy_to(ScaLAPACKMatrix<NumberType> &B,
+                                     const std::pair<unsigned int,unsigned int> &offset_A,
+                                     const std::pair<unsigned int,unsigned int> &offset_B,
+                                     const std::pair<unsigned int,unsigned int> &submatrix_size) const
+{
+  //submatrix is empty
+  if (submatrix_size.first == 0 || submatrix_size.second == 0)
+    return;
+
+  //range checking for matrix A
+  Assert (offset_A.first<(unsigned int)(n_rows-submatrix_size.first+1),
+          ExcIndexRange(offset_A.first,0,n_rows-submatrix_size.first+1));
+  Assert (offset_A.second<(unsigned int)(n_columns-submatrix_size.second+1),
+          ExcIndexRange(offset_A.second,0,n_columns-submatrix_size.second+1));
+
+  //range checking for matrix B
+  Assert (offset_B.first<(unsigned int)(B.n_rows-submatrix_size.first+1),
+          ExcIndexRange(offset_B.first,0,B.n_rows-submatrix_size.first+1));
+  Assert (offset_B.second<(unsigned int)(B.n_columns-submatrix_size.second+1),
+          ExcIndexRange(offset_B.second,0,B.n_columns-submatrix_size.second+1));
+
+  //Currently, copying of matrices will only be supported if A and B share the same MPI communicator
+  int ierr, comparison;
+  ierr = MPI_Comm_compare(grid->mpi_communicator,B.grid->mpi_communicator,&comparison);
+  Assert (comparison == MPI_IDENT,ExcMessage("Matrix A and B must have a common MPI Communicator"));
+
+  /*
+   * The routine pgemr2d requires a BLACS context resembling at least the union of process grids
+   * described by the BLACS contexts of matrix A and B
+   */
+  int union_blacs_context = Csys2blacs_handle(this->grid->mpi_communicator);
+  const char *order = "Col";
```

Comment:
i suppose this should be consistent to what we chose in `ProcessGrid` class? If so, maybe add a comment so that we know that one has too keep them consistent